experimental/torch_xla2/torch_xla2/functions.py

```diff
@@ -0,0 +1,29 @@
+"""Tensor constructor overrides"""
+import warnings
+
+import torch
+import jax.numpy as jnp
+from torch_xla2 import tensor
+
+fns = {
+  torch.tensor: jnp.array,
+  # torch.ones: jnp.ones,
+  # torch.zeros: jnp.zeros,
+  # torch.arange: jnp.arange,
+  # torch.linspace: jnp.linspace,
+  # torch.logspace: jnp.logspace,
+  # torch.empty: jnp.empty,
+  # torch.eye: jnp.eye,
+  # torch.full: jnp.full,
+}
+
+class XLAFunctionMode(torch.overrides.TorchFunctionMode):
+  def __torch_function__(self, func, types, args=(), kwargs=None):
+    jax_func = fns.get(func)
+    if not jax_func:
+      raise NotImplementedError(f'No jax function found for {func.__name__}')
+
+    if kwargs:
+      warnings.warn(f'kwargs not implemented for {kwargs}')
+
+    return tensor.wrap(jax_func(*args))
```

Comment:
args need to be unpacked with `args = tensor.unwrap(args)`. for tensor constructors it doesn't matter because likely it's input are not other tensors; for other torch functions it does

Reply:
Looking more closely, many of the constructors are explicitly `int`s and not another tensor. So in the function dispatcher, I think we can leave the inputs alone, and then I can add a decorator like `@convert_dtype` to handle any cases where we can end up taking other `Tensor`s.

examples: https://pytorch.org/docs/stable/generated/torch.full.html and https://pytorch.org/docs/stable/generated/torch.ones.html